### ISSUE TITLE
NonMatchingSplitsSizesError on Prepare Step 

### ISSUE THREAD

**Issue by ArjunRamaswami22** [mila-iqia/milabench#184]
I seem to have the following error with these particular benchmarks `opt-1_3b` and `opt-6_7b`, when running v0.0.7 on a single node with 4 GPUs: 
 
```Shell
NonMatchingSplitsSizesError: [{'expected': SplitInfo(name='test',           
num_bytes=1305088, num_examples=4358, shard_lengths=None,                   
dataset_name=None), 'recorded': SplitInfo(name='test', num_bytes=5176698,   
num_examples=17432, shard_lengths=None, dataset_name='parquet')} ...
```

Trying to run these benchmarks using the manual (non-container) setup given in the documentation. Is this a config problem? I am using the `standard.yaml` config here. The logs are attached below.

[opt-1_3b.stderr.txt](https://github.com/mila-iqia/milabench/files/13922437/opt-1_3b.stderr.txt)
[opt-6_7b.stderr.txt](https://github.com/mila-iqia/milabench/files/13922438/opt-6_7b.stderr.txt)

**Comment by Delaunay** [comment]
The error happens when the downloads are being verified.
The dataset has smaller samples than expected, maybe the download was not entirely successful.

The containers are provided as a way to limit the number of steps third party have to execute
to limit the number of issue that can arise.

```
[
{
'expected': SplitInfo(name='test', num_bytes=1305088, num_examples=4358, shard_lengths=None, dataset_name=None), 
'recorded': SplitInfo(name='test', num_bytes=5176698,  num_examples=17432, shard_lengths=None, dataset_name='parquet')
}, 

{
'expected': SplitInfo(name='train', num_bytes=546500949,  num_examples=1801350, shard_lengths=None, dataset_name=None), 
'recorded': SplitInfo(name='train', num_bytes=1113622699, num_examples=3676136,  shard_lengths=[1650675, 1661350,364111], dataset_name='parquet')
}, 

{
'expected': SplitInfo(name='validation', num_bytes=1159288, num_examples=3760, shard_lengths=None, dataset_name=None), 
'recorded': SplitInfo(name='validation', num_bytes=4607450, num_examples=15040, shard_lengths=None,dataset_name='parquet')
}
]
```

**Comment by ArjunRamaswami22** [comment]
I see, I tried running the docker container using enroot. I receive this strange error when I try to mount the container:

```
[xxx@compn04 ~]$ enroot start --mount .:/myfiles mila-0.0.7
Traceback (most recent call last):
 File "/usr/local/lib/python3.10/dist-packages/pynvml/nvml.py", line 1798, in _LoadNvmlLibrary
   nvmlLib = CDLL("libnvidia-ml.so.1")
 File "/usr/lib/python3.10/ctypes/__init__.py", line 374, in __init__
   self._handle = _dlopen(self._name, mode)
OSError: libnvidia-ml.so.1: cannot open shared object file: No such file or directory
 
During handling of the above exception, another exception occurred:
 
Traceback (most recent call last):
 File "/usr/local/lib/python3.10/dist-packages/voir/instruments/gpu/cuda.py", line 57, in __init__
   nvmlInit()
 File "/usr/local/lib/python3.10/dist-packages/pynvml/nvml.py", line 1770, in nvmlInit
   nvmlInitWithFlags(0)
 File "/usr/local/lib/python3.10/dist-packages/pynvml/nvml.py", line 1753, in nvmlInitWithFlags
   _LoadNvmlLibrary()
 File "/usr/local/lib/python3.10/dist-packages/pynvml/nvml.py", line 1800, in _LoadNvmlLibrary
  _nvmlCheckReturn(NVML_ERROR_LIBRARY_NOT_FOUND)
 File "/usr/local/lib/python3.10/dist-packages/pynvml/nvml.py", line 833, in _nvmlCheckReturn
   raise NVMLError(ret)
pynvml.nvml.NVMLError_LibraryNotFound: NVML Shared Library Not Found
 
The above exception was the direct cause of the following exception:
 
Traceback (most recent call last):
 File "/usr/local/bin/milabench", line 8, in <module>
   sys.exit(main())
 File "/milabench/milabench/milabench/cli.py", line 42, in main
   sys.exit(run_cli(Main, argv=argv))
 File "/usr/local/lib/python3.10/dist-packages/coleo/cli.py", line 628, in run_cli
   return call(opts=opts, args=args)
 File "/usr/local/lib/python3.10/dist-packages/coleo/cli.py", line 587, in thunk
   result = fn(*args)
 File "/milabench/milabench/milabench/cli.py", line 323, in run
   init_arch()
 File "/milabench/milabench/milabench/cli.py", line 154, in init_arch
   return select_backend(arch)
 File "/usr/local/lib/python3.10/dist-packages/voir/instruments/gpu/__init__.py", line 91, in select_backend
   DEVICESMI = backend.DeviceSMI()
 File "/usr/local/lib/python3.10/dist-packages/voir/instruments/gpu/cuda.py", line 60, in __init__
   raise NotAvailable() from err
voir.errors.NotAvailable
```

Is there some environment variable that has to be passed when starting the container for the right backend to be selected?

I created the container from the image like this:
`enroot create --name mila-0.0.7 mila-iqia+milabench+cuda-v0.0.7.sqsh`

**Comment by Delaunay** [comment]
Docker image with CUDA runs with docker + [nvidia container toolkit](https://github.com/NVIDIA/nvidia-container-toolkit) / [libnvidia-container](https://github.com/NVIDIA/libnvidia-container)

I am believe those errors are probably related to that.
Seems like the libraries required to interface with the GPUs are not exposed.

**Comment by Delaunay** [comment]
Fixed in https://github.com/mila-iqia/milabench/pull/187 released in v0.0.8
